taskcollector/TaskCollector.cpp

```diff
@@ -85,44 +82,51 @@ void FreeImageErrorHandler(FREE_IMAGE_FORMAT fif, const char *message) {
 // ----------------------------------------------------------
 
 class TaskCollector {
-	
+
 protected:
-	std::list<FIBITMAP *> chunks;
-	/*std::queue<FIBITMAP *> alphaChunks; */
-	
+	std::vector<bitmap_t> chunks;
+	std::vector<bitmap_t> alphaChunks;
 
 public:
-	
+	TaskCollector() = default;
+	TaskCollector(const TaskCollector&) = default;
+	//TaskCollector(TaskCollector &&) = default;	// for some reason visual think it's incorrect constructor
+	~TaskCollector()
+	{
+		std::for_each(chunks.begin(), chunks.end(), [](bitmap_t el){ FreeImage_Unload(el.get()); });
+		std::for_each(alphaChunks.begin(), alphaChunks.end(), [](bitmap_t el){ FreeImage_Unload(el.get()); });
+	}
+
 	bool addImgFile(const char* pathName, int flag = 0)  {
-		FIBITMAP *img = GenericLoader(pathName, flag);
-		if (img == NULL) 
+		auto img = GenericLoader(pathName, flag);
+		if (img == NULL)
 			return false;
 		chunks.push_back(img);
 		return true;
 	};
 
 	bool addAlphaFile(const char* pathName, int flag = 0) {
-		FIBITMAP *img = GenericLoader(pathName, flag);
+		auto img = GenericLoader(pathName, flag);
 		if (img == NULL)
 			return false;
-		alphaChunks.push(img);
+		alphaChunks.push_back(img);

```

Comment:
I was checking my changes on debug build and I didn't notice this failure. This time I've checked both: debug and release build and nothing crashes on my PC